docs/OTP2-MigrationGuide.md

```diff
@@ -32,6 +32,7 @@ These properties changed names from:
  - `searchWindow` Limit the departure window or arrival window for the routing search.
  - `boardSlackByMode` How much time boarding a vehicle takes for each given mode.
  - `alightSlackByMode` How much time alighting a vehicle takes for each given mode.
+ - `modes` The REST API is unchanged, but is mapped into a new structure in the RoutingRequest. This means not all combinations of non-transit modes that was available in OTP1 is available in OTP2.
```

Comment:
> There is 5 classes in OTP2 (after this is merged) that represent modes in slightly different scopes with overlapping types - like TransitMode and TraverseMode, witch both include mode like BUS. I think this is confusing. I will not insist that we change this, just express a bit of concern about it.
> 
> If we make two sub-types for itinerary Leg, a TransitLeg and a StreetLeg, then _maybe_ we could get rid of the `TraverseMode`. But are modes really static structures? In StreetMode there is more than one CAR and BIKE mode, to me this is a signal that we are missing something? The set (RENTED, BIKE, FOR_ACCESS) is not a mode. The mode is BIKE, the fact that it is rented and used for access tells us that the next mode transition is going to happen in a place where we can park the bike. Modes are in many contexts a hierarchical structure (e.g. NeTEx), so maybe it would be better to represent them in another way. I do not have a complete overview of what we use modes for - so it is hard to suggest something concrete.

I agree that StreetMode is not a mode in the same sense as the others, but I'm struggling to find a good name for it. In some ways it represents a state-machine with modes and allowed transitions.